### DISCUSSION TITLE
Accessing the whatsapp database on android

### DISCUSSION THREAD

**Discussion by mosheduminer** [karlicoss/HPI#366] [General]
Hi!

I discovered HPI not that long ago (via this [HN comment](https://news.ycombinator.com/item?id=38821803)) and was inspired by it. I would like to create an app (intended for myself and friends) to better search through my whatsapp groups (where a lot of information goes back and forth) as the built in search has proved insufficient (especially in terms of UX). I saw that there is a python module for this (which I would like to use as prior art), but I'm stuck the point of getting access to the database, which according to [this code comment](https://github.com/karlicoss/HPI/blob/87a8a7781bfb335dfc139a62ca0e04bb8045087f/my/whatsapp/android.py#L2) should be in `Android/data`, but I'm unable to see anything in there using the google file explorer (for safety reasons I assume). I see the backup databases in `Android/media` which would be fine for my purposes, but they're encrypted (despite me not having encrypted backups turned on in my settings?) and I can't figure out how to unencrypt them (I assume the key is in `Android/data` somewhere).

So, I'm wondering how you get ahold of the whatsapp database?

**Comment by karlicoss** [comment]
Hey! You need root on your phone unfortunately. The data used in HPI is taken from /data/data/com.whatsapp, don't think there is anything useful in media folder.

**Reply by PatMyron** [reply]
While [Android may now be the most popular OS](https://gs.statcounter.com/os-market-share/), generally curious about tradeoffs of the pattern of accessing service data through Android (especially requiring root) since OS-agnosticism seems preferable if feasible:

[`bumble/android.py`](https://github.com/karlicoss/HPI/blob/master/src/my/bumble/android.py)
[`tinder/android.py`](https://github.com/karlicoss/HPI/blob/master/src/my/tinder/android.py)
[`twitter/android.py`](https://github.com/karlicoss/HPI/blob/master/src/my/twitter/android.py)
[`whatsapp/android.py`](https://github.com/karlicoss/HPI/blob/master/src/my/whatsapp/android.py)
[`instagram/android.py`](https://github.com/karlicoss/HPI/blob/master/src/my/instagram/android.py)
[`fbmessenger/android.py`](https://github.com/karlicoss/HPI/blob/master/src/my/fbmessenger/android.py)
[`google/maps/android.py`](https://github.com/karlicoss/HPI/blob/master/src/my/google/maps/android.py)

**Reply by karlicoss** [reply]
Oh I'd definitely much prefer not to do that! But for this data there are no other options really.

**Reply by PatMyron** [reply]
Most of those services offer GDPR/CCPA-esque account data export requests:
https://github.com/bionic/how-to-export-personal-data/pull/5
and at least a couple of them vended that data within a day:
https://github.com/bionic/how-to-export-personal-data/issues/6
